packages/runtime/src/core.ts

```diff
@@ -520,11 +520,26 @@ export class FederationHost {
     moduleOptions: ModuleOptions;
     remoteMatchInfo: LoadRemoteMatch;
   }> {
-    const loadRemoteArgs = await this.hooks.lifecycle.beforeRequest.emit({
-      id,
-      options: this.options,
-      origin: this,
-    });
+    let loadRemoteArgs;
+
+    try {
+      loadRemoteArgs = await this.hooks.lifecycle.beforeRequest.emit({
+        id,
+        options: this.options,
+        origin: this,
+      });
+    } catch (error) {
+      loadRemoteArgs = await this.hooks.lifecycle.errorLoadRemote.emit({
+        id,
+        error,
+        from: 'runtime',
```

Comment:
should we include another option in the error hook?
like hook: 'beforeRequest'

this way errorLoadRemote hook would know what type of args to return based on what lifecycle throw error? 

if it throw on loadRemote, then it ca return fallback Module
if it throw on beforeRequest, then it can return the args that should return for beforeRequest hook.

Reply:
yeah, i think it's good !